### DISCUSSION TITLE
Concatenate lists with if statement

### DISCUSSION THREAD

**Discussion by PierreAdrienDelisle** [cuelang/cue#623] [Q&A] [answered]
Hello, sorry if the question seems stupid but I had a problem understanding how to solve a problem I had. I want to concatenate two lists and the values of those lists depends on a if statement like this :
```
if lb != null {
  let lbDep = ["value"]
}
if ub != null {
  let ubDep = ["secVal"]
}
dep: lbDep+ ubDep

```

The problem here is that they (lbDep/ubDep) arent't in the scope of the var "dep".
But if I initialize them with [] (empty array), I get a "conflicting values" error. 
```
lbDep: []
if lb != null {
  lbDep = ["value"]
}
ubDep:[]
if ub != null {
  ubDep = ["secVal"]
}
dep: lbDep+ ubDep

```

How is it possible to overcome this or re-define an existing value? 
Thanks!

**Comment by verdverm** [comment]
- try initializing with `lbDep: [...]`. `[]` is an empty list while `[...]` is a list of unknown elements
- a `list.FlattenN` in the stdlib too

**Reply by PierreAdrienDelisle** [reply]
I was able to do what I wanted thanks to you guys ! I didn't know for the [...] as I'm a beginner 
For people having the same struggle here it is:
```
lb: null
ub: "test"

lbDep: [...]
if lb != null {
  lbDep: ["value"]
}
ubDep:[...]
if ub != null {
  ubDep: ["secVal"]
}
dep: lbDep + ubDep

```

**Comment by mpvl** [comment, marked as answer]
On tip, at least, you can do this:
```
dep: [
    if lb != null {
        "value"
    },
    if ub != null {
        "secVal"
    },
]
```

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/623.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
